(inconsequential) error when running bundlejs

```julia
julia> WebIO.bundlejs(watch=true)

Webpack is watching the files…

Hash: 283f6e34baa0f1142700
Version: webpack 2.2.1
Time: 306ms
          Asset    Size  Chunks             Chunk Names
webio.bundle.js  121 kB       0  [emitted]  WebIO
   [0] ./basics/webio.js 3.05 kB {0} [built]
   [1] (webpack)/buildin/global.js 509 bytes {0} [built]
   [2] ./basics/node.js 8.27 kB {0} [built]
   [3] ./~/base64-js/index.js 3.48 kB {0} [built]
   [4] ./~/buffer/index.js 48.6 kB {0} [built]
   [5] ./~/ieee754/index.js 2.05 kB {0} [built]
   [6] ./~/isarray/index.js 132 bytes {0} [built]
   [7] ./~/process/browser.js 5.3 kB {0} [built]
   [8] ./~/systemjs/dist/system.js 46.4 kB {0} [built]
   [9] ./index.js 74 bytes {0} [built]

ERROR in ./~/systemjs/dist/system.js
Module not found: Error: Can't resolve 'fs' in '/home/shashi/.julia/v0.5/WebIO/assets/node_modules/systemjs/dist'
 @ ./~/systemjs/dist/system.js 4:11220-11233
 @ ./basics/node.js
 @ ./index.js
^CERROR: InterruptException:
 in run at ./process.jl:592 [inlined]
 in (::WebIO.##27#28{Bool})() at /home/shashi/.julia/v0.5/WebIO/src/devsetup.jl:18
 in cd(::WebIO.##27#28{Bool}, ::String) at ./file.jl:59
 in (::WebIO.#kw##bundlejs)(::Array{Any,1}, ::WebIO.#bundlejs) at ./<missing>:0
 in eval_user_input(::Any, ::Base.REPL.REPLBackend) at ./REPL.jl:64
 in macro expansion at ./REPL.jl:95 [inlined]
 in (::Base.REPL.##3#4{Base.REPL.REPLBackend})() at ./event.jl:68
```

--- a/assets/basics/webio.js
+++ b/assets/basics/webio.js
@@ -1,4 +1,5 @@
 var contexts = {};
+var obscontexts = {};
 
 function makeWidget(id, data, sendCallback, dom, handlers, observables)
 {
@@ -14,6 +15,16 @@ function makeWidget(id, data, sendCallback, dom, handlers, observables)
     }
 
     contexts[id] = ctx;
+    if (observables){
+        Object.keys(observables).forEach(function setobscontext(name){
+            var o = observables[name]
+            if (typeof obscontexts[o.id] === "undefined"){
+                obscontexts[o.id] = []
+            }
+            //obname is the name of the observable in this context
+            obscontexts[o.id].push({ctx: ctx, obname: name})
+        })
+    }
 
     return ctx;
 }
@@ -57,7 +68,7 @@ function dispatch(msg)
         console.log(fs)
         for (var i=0, l=fs.length; i<l; i++) {
             var f = fs[i]
-            f.apply(ctx, [msg.data, false]) // false for "not client-side"
+            f.call(ctx, msg.data, false) // false for "not client-side"
         }
     }
 }
@@ -93,30 +104,39 @@ function send(ctx, cmd, data)
 }
 
 function setval(ob, val) {
-    var ctx = contexts[ob.context];
-    var x = ctx.observables[ob.name]
-    x.value = val;
-
-    if (ctx.handlers[ob.name] !== undefined) {
-        var fs = ctx.handlers[ob.name];
-        for (var i=0, l=fs.length; i<l; i++) {
-            var f = fs[i]
-            f.apply(ctx, [val, true]) // true for "client-side"
+    var allcontexts = obscontexts[ob.id]
+    var synced_julia = false
+    allcontexts.forEach(function propagate_to_other_contexts(octxinfo){
+        var ctx = octxinfo.ctx
+        var name = octxinfo.obname // the name of the observable in octx
+        var x = ctx.observables[name]
+        if (val === x.value || (val !== val && x.value !== x.value)) {
+            // adapted from Vue.js reactiveSetter, avoids calling handlers if value
+            // is unchanged. The second check is for values like NaN, since, e.g.,
+            // NaN !== NaN
+            return
         }
-    }
+        x.value = val
+        if (typeof ctx.handlers[name] !== "undefined") {
+            var fs = ctx.handlers[name];
+            for (var i=0, l=fs.length; i<l; i++) {
+                var f = fs[i]
+                f.call(ctx, val, true) // true for "client-side"
+            }
+        }
 
-    if (x.sync) {
-        WebIO.send(ctx, ob.name, val);
-    }
+        // sync the observable's value to julia if `sync` is true in any context
+        if (x.sync && !synced_julia) {
+            WebIO.send(ctx, name, val);
+            synced_julia = true
+        }
+    })
 }
 
 function getval(ob) {
     var ctx = contexts[ob.context];
     var x = ctx.observables[ob.name]
     x.value = val;
-    if (x.sync) {
-        WebIO.send(ctx, ob.name, val);
-    }
 }
 
 function message(ctx, cmd, data)

--- a/assets/webpack.config.js
+++ b/assets/webpack.config.js
@@ -1,8 +1,11 @@
 module.exports = {
   entry: {
-      WebIO: './index.js'
+    WebIO: './index.js'
   },
   output: {
-    filename: 'webio.bundle.js'       
+    filename: 'webio.bundle.js'
+  },
+  node: {
+    fs: 'empty'
   }
 };

--- a/src/node.jl
+++ b/src/node.jl
@@ -105,7 +105,7 @@ function JSON.lower(n::Node)
         "type" => "node",
         "nodeType" => nodetype(n),
         "instanceArgs" => JSON.lower(n.instanceof),
-        "children" => children(n),
+        "children" => children_to_show(n),
         "props" => props(n),
     )
 end
@@ -113,9 +113,21 @@ end
 # Dict to store callbacks when displaying a Node
 const showcbs = Dict{Node, Function}()
 
+"""
+Runs show callbacks on child nodes if any are defined. This will be called
+when `JSON.lower(n::Node)` runs, and thus will end up being called recursively
+on all descendent Nodes from `n`.
+"""
+function children_to_show(n::Node)
+    map!(Vector{Union{Node, String}}(length(children(n))), children(n)) do x
+        x in keys(showcbs) && (x = showcbs[x]())
+        x
+    end
+end
+
 function Base.show(io::IO, m::MIME"text/html", x::Node)
     id = newid("node")
-    x in keys(showcbs) && (x = showcbs[x](io, id))
+    x in keys(showcbs) && (x = showcbs[x](id))
     write(io, """<div id='$id'></div>
                  <script>WebIO.mount('$id', '#$id',""")
     jsexpr(io, x)

--- a/src/syntax.jl
+++ b/src/syntax.jl
@@ -152,7 +152,8 @@ function jsexpr(io, o::Observable)
 
     obsobj = Dict("type" => "observable",
                   "context" => ctx.id,
-                  "name" => name)
+                  "name" => name,
+                  "id" => obsid(o))
 
     jsexpr(io, obsobj)
 end

--- a/src/syntax.jl
+++ b/src/syntax.jl
@@ -24,7 +24,7 @@ function makedom(tag, props)
         ns, t = split(string(tag), ":")
         DOM(Symbol(ns), Symbol(t))
     else
-        DOM(:xhtml, tag)
+        DOM(:html, tag)
     end
     function dom(args...; kwargs...)
         n = Node(d, args...)(Dict(kwargs))

--- a/src/widget.jl
+++ b/src/widget.jl
@@ -24,7 +24,7 @@ Fields:
 - `dependencies`: An array of js/html/css assets to load
   before rendering the contents of a context.
 """
-immutable Widget
+struct Widget
     id::AbstractString
     outbox::Channel
     observs::Dict{String, Tuple{Observable, Union{Void,Bool}}} # bool marks if it is synced
@@ -48,6 +48,14 @@ function Widget(
 
     contexts[id] = Widget(id, outbox, observs, dependencies, jshandlers)
 end
+
+"""
+`set_id(w::Widget, newid)`
+Returns a new Widget with id=`newid` and all other props taken from `w`
+"""
+set_id(w::Widget, newid) =
+  Widget(newid, outbox=w.outbox, observs=w.observs,
+         dependencies= w.dependencies, jshandlers=w.jshandlers)
 
 function Observables.on(f, w::Widget, cmd)
     listener, _ = Base.@get! w.observs cmd (Observable{Any}(w, cmd, nothing), nothing)
@@ -112,7 +120,8 @@ function lowerobserv(ob_)
         sync = any(f-> !isa(f, Backedge), ob.listeners)
     end
     Dict("sync" => sync,
-         "value" => ob[])
+         "value" => ob[],
+         "id" => obsid(ob))
 end
 
 function lowerdeps(x::String)

